Display a portal warning on the overview of trashed documents

Rotonen opened the pull request:
* Adding a portal warning onto the tabbedview of a document or a mail
  * Only if the request is a 'simple get'
* Adding a warning to bumblebee overlays for documents and mails
* Added a versioned attribute to the versioned overlay as the adapter loses the context proper

Document overview:
![Screenshot 2019-03-25 at 15 46 29](https://user-images.githubusercontent.com/935317/54929129-60e46880-4f15-11e9-9602-30af56d8c2c7.png)

Document overview overlay:
![Screenshot 2019-03-25 at 15 46 40](https://user-images.githubusercontent.com/935317/54929196-7a85b000-4f15-11e9-8ac8-0bf1d54ea697.png)

Document version listing overlay of current version:
![Screenshot 2019-03-25 at 15 47 05](https://user-images.githubusercontent.com/935317/54929215-81acbe00-4f15-11e9-8157-d0da59a8212f.png)

Document version listing overlay of old version:
![Screenshot 2019-03-25 at 15 47 12](https://user-images.githubusercontent.com/935317/54929303-9f7a2300-4f15-11e9-8700-fcf086ef9da2.png)

Closes #5481

deiferni commented on opengever/document/browser/tabbed.py:
```diff
@@ -8,6 +11,24 @@ class DocumentTabbedView(TabbedView):
     uploadbox, to allow drag and drop replacing of a file.
     """
 
+    def __init__(self, context, request):
```
why not __call__, __init__ may be a bit early in the lifecycle 🤔.

Rotonen replied:
I recoiled from doing that as the `__call__` of the parent is overriding the `__call__` of the Five `BrowserView` in a weird way and I've never had a side effect free ride when interfacing with a `ViewPageTemplateFile` bouncing around in the publisher.

https://github.com/4teamwork/ftw.tabbedview/blob/649f6f3e05e1241374cd05f1ee2bdeadd12affad/ftw/tabbedview/browser/tabbed.py#L37

I'll give it a shot.

Rotonen replied:
And the publisher loses the context if I try that. Sticking with `__init__` here.

Rotonen commented:
* Added missing overlay functionality
* Also now tagging mails

Rotonen commented:
@deiferni thank you. I completely missed the overlay adapter mangles the overlay context on versioned documents.

That seems to be a side effect of https://github.com/4teamwork/opengever.core/pull/1967, but I'm not certain if it's intentional. In any case one can work around this as the original implementation has worked around the retrival of the version ID as well.

* Fixed version listing overlays
* Made the overlay warning position consistent

Also added screenshots to the main PR body.

deiferni commented on opengever/document/locales/de/LC_MESSAGES/opengever.document.po:
```diff
@@ -694,6 +694,11 @@ msgstr "Dieses Dokument kann nicht zu einem PDF konvertiert werden"
+#. Default: "This document is trashed."
+#: ./opengever/document/browser/tabbed.py
+msgid "warning_trashed"
+msgstr "Das Dokument ist im Papierkorb."
```
sorry for mentioning this only now but i think the translation could be improved slightly:

Das Dokument ist im Papierkorb -> Dieses Dokument befindet sich im Papierkorb.

Same for the mails.

deiferni commented on opengever/mail/locales/de/LC_MESSAGES/opengever.mail.po:
```diff
@@ -244,3 +244,8 @@ msgstr "E-Mails"
+#. Default: "This mail is trashed."
+#: ./opengever/mail/browser/tabbed.py
+msgid "warning_trashed"
+msgstr "Das E-Mail ist im Papierkorb."
```
@Rotonen unfortunately this translation was not updated correctly as requested in https://github.com/4teamwork/opengever.core/pull/5505#discussion_r268990657
